iceoryx_hoofs/source/cxx/helplets.cpp

```diff
@@ -21,27 +21,33 @@ namespace iox
 {
 namespace cxx
 {
+// AXIVION DISABLE STYLE AutosarC++19_03-A5.2.4 : allow reinterpret_cast for the whole file
 void* alignedAlloc(const uint64_t alignment, const uint64_t size) noexcept
 {
     // -1 == since the max alignment addition is alignment - 1 otherwise the
     // memory is already aligned and we have to do nothing
     // low-level memory management
     // NOLINTNEXTLINE(cppcoreguidelines-owning-memory, cppcoreguidelines-no-malloc, hicpp-use-auto)
-    uint64_t memory = reinterpret_cast<uint64_t>(malloc(size + alignment + sizeof(void*) - 1));
+    // AXIVION Next Line AutosarC++19_03-M5.2.9 : Pointer conversion is safe here.
+    // AXIVION Next Line AutosarC++19_03-A18.5.1 : Dynamic memory management function is required.
+    const uint64_t memory = reinterpret_cast<uint64_t>(malloc((size + alignment + sizeof(void*)) - 1));
```

Comment:
Could you maybe change this to
```suggestion
    const uint64_t memory = reinterpret_cast<uint64_t>(malloc((sizeof(void*) + alignment + size) - 1));
```
This makes it a bit clearer how the memory is used and prevents to attribute the `-1` to `sizeof(void*)`

Reply:
Could we introduce here a global constexpr called `constexpr uint64_t IOX_MEMORY_ADDRESS_SIZE = sizeof(void*)`. Then the code would be more readable since the intention would be clearer.